src/app/shared/components/tooltip/tooltip.component.ts

```diff
@@ -0,0 +1,61 @@
+import { AfterViewInit, ChangeDetectorRef, Component, ElementRef, Input, OnDestroy, ViewChild } from '@angular/core'
+import { Observable, Subject, takeUntil } from 'rxjs'
+
+@Component({
+  selector: 'sofie-tooltip',
+  templateUrl: './tooltip.component.html',
+  styleUrls: ['./tooltip.component.scss'],
+})
+export class TooltipComponent implements OnDestroy, AfterViewInit {
+  @Input() public tooltipElementHoverMouseEventObservable: Observable<MouseEvent | undefined>
```

Comment:
I am not that familiar with angular practices, but passing in an observable as an input seems a bit wrong. It is passing a reactive element (the observable) into a reactive property (the input variable).
And if the input observable changes, we have a dangling subscription and we won't subscribe to the new observable, potentially giving weird behavior.

My suggestion will be to rely on the `@Input()` parameter and instead of passing in a `MouseEvent`, then just pass in the minimal data the component needs, namely a position. Then we can use `ngOnChange` to detect changes. In the example below I called the attribute `pagePosition` to clarify what coordinate system is used.

```typescript
interface Position {
  readonly x: number
  readonly y: number
}

export class TooltipComponent implements .... {
  @Input() public pagePosition?: Position

  public ngOnChange(changes: SimpleChanges): void {
    if ('pagePosition' in changes) {
      this.updateVisibility()
    }
  }
}
```

Reply:
Done